.github/actions/download-pr-data-artifact/action.yaml

```
@@ -18,18 +18,25 @@ runs:
               repo: context.repo.repo,
               run_id: context.payload.workflow_run.id,
           });
+
           let matchArtifact = allArtifacts.data.artifacts.filter((artifact) => {
             return artifact.name == "pr_number"
           })[0];
-          let download = await github.rest.actions.downloadArtifact({
-              owner: context.repo.owner,
-              repo: context.repo.repo,
-              artifact_id: matchArtifact.id,
-              archive_format: 'zip',
-          });
-          let fs = require('fs');
-          fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/pr_number.zip`, Buffer.from(download.data));
-    - name: 'Unzip artifact'
+
+          # Check if matchArtifact is defined
+          if (matchArtifact && marginArtifact.id) {
+            let download = await github.rest.actions.downloadArtifact({
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                artifact_id: matchArtifact.id,
+                archive_format: 'zip',
+            });
+            let fs = require('fs');
+            fs.writeFileSync(`${process.env.GITHUB_WORKSPACE}/pr_number.zip`, Buffer.from(download.data));
+          } else {
+            console.log('No artifact named "pr_number" found');
+          }
```

Should this fail if the PR number is not set? what should the behavior be?